collector/container/konflux.Dockerfile

```
@@ -0,0 +1,196 @@
+    && cp -a /builder/third_party third_party \
+    && cp -a /kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION \
+    && cp -a /CMakeLists.txt CMakeLists.txt
+
+# WITH_RHEL_RPMS controls for dependency installation, ie if they were already installed as RPMs.
+# Setting the value to empty will cause dependencies to be downloaded from repositories or accessed in submodules and compiled.
+# TODO(ROX-20651): Set ENV WITH_RHEL_RPMS=true when RHEL RPMs can be installed to enable hermetic builds.
+# ENV WITH_RHEL_RPMS=true
+
+# Build with gperftools (DISABLE_PROFILING=OFF) only for supported
+# architectures, at the moment x86_64 only
+RUN ./builder/install/install-dependencies.sh && \
+    if [[ "$(uname -m)" == "x86_64" ]];   \
+        then DISABLE_PROFILING="OFF";   \
+        else DISABLE_PROFILING="ON";    \
+    fi ; \
+    cmake -S ${SRC_ROOT_DIR} -B ${CMAKE_BUILD_DIR} \
+           -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
+           -DDISABLE_PROFILING=${DISABLE_PROFILING} \
+           -DCOLLECTOR_APPEND_CID=${COLLECTOR_APPEND_CID} \
+           -DUSE_VALGRIND=${USE_VALGRIND} \
+           -DADDRESS_SANITIZER=${ADDRESS_SANITIZER} \
+           -DTRACE_SINSP_EVENTS=${TRACE_SINSP_EVENTS} && \
+    cmake --build ${CMAKE_BUILD_DIR} --target all -- -j "${NPROCS:-2}" && \
+    ctest -V --test-dir ${CMAKE_BUILD_DIR} && \
+    strip -v --strip-unneeded "${CMAKE_BUILD_DIR}/collector/collector"
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS support-packages-downloader
+
+WORKDIR /staging
+
+COPY kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION
+COPY collector/container/scripts/download-support-package.sh download-support-package.sh
+
+RUN ./download-support-package.sh
+
+# Do NOT use follow_tag here, as we do not need or want collector to be rebuilt
+# with each drivers build (which may become very frequent)
+FROM brew.registry.redhat.io/rh-osbs/rhacs-drivers-build-rhel8:0.1.0 AS drivers-build
+# FROM registry-proxy.engineering.redhat.com/rh-osbs/rhacs-drivers-build-rhel8:0.1.0 AS drivers-build
+
+# TODO(ROX-20312): we can't pin image tag or digest because currently there's no mechanism to auto-update that.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS unpacker
+
+RUN microdnf install -y unzip findutils
+WORKDIR /staging
+
+COPY --from=support-packages-downloader /staging/support-pkg.zip /staging/
+COPY kernel-modules/MODULE_VERSION MODULE_VERSION.txt
+# Creating this directory ensures the scratch build with dummy support-pkg.zip will not fail.
+RUN mkdir -p "/staging/kernel-modules/$(cat MODULE_VERSION.txt)"
+# First, unpack upstream support package, only on x86_64
+RUN if [[ "$(uname -m)" == x86_64 ]]; then unzip support-pkg.zip ; fi
+# Fail non-scratch build if there were no drivers matching the module version.
+RUN if [[ "$(uname -m)" == x86_64 && "$(ls -A /staging/kernel-modules/$(cat MODULE_VERSION.txt))" == "" && "$(unzip -Z1 support-pkg.zip)" != "dummy-support-pkg" ]] ; then \
```

Let's remove "scratch" build things here too.

```suggestion

RUN if [[ "$(uname -m)" == x86_64 && "$(ls -A /staging/kernel-modules/$(cat MODULE_VERSION.txt))" == "" ]] ; then \
```